### DISCUSSION TITLE
How do I reuse layout with background image?

### DISCUSSION THREAD

**Discussion by t-zander** [vercel/next.js#20503] [Help] [answered]
Hello there.
I have an authentication page, which is an image and two tabs: for sign in and sign up respectively.
![Auth page](https://user-images.githubusercontent.com/34350867/103159354-20440e00-47d1-11eb-975c-3c63459ca8f4.png)

I used to solve this by creating AuthLayout component, which holds the top image and two tabs.

Then I created `auth` folder which has `sign-in.tsx` and `sign-up.tsx` files. 

Here is a relevant part of AuthLayout
```
return (
    <Box className={classes.wrapper}>
      <Box className={classes.topBar}>
        <Image src="/auth-page.png" layout="fill" objectFit="cover" />
      </Box>
      <Box className={classes.authWrapper}>
        <Navigation />
        {children} / either sign in or sign up part
      </Box>
    </Box>
  );
```

When I am switching between tabs, layout is re rendered, thus image component is re rendered too. Thus, image is flickering.
How would I go about this? In simple React, I would create common component, and then would make my sign in and sign up tabs just a nested <Route /> components. But here, I cannot do it like this. 
Any help is much appreciated.

**Comment by jamesmosier** [comment, marked as answer]
Hi @t-zander. You can use a [custom `App.js`](https://nextjs.org/docs/advanced-features/custom-app) for this, although it would be for your entire application. This would then be your "layout" component and would not re-render when navigating between pages (assuming your are using [next/link](https://nextjs.org/docs/api-reference/next/link) to navigate between pages).

If you just wanted this for your login and register pages, you could still use a custom `App.js` and set an instance variable for each:

```js
// pages/app.tsx
function MyApp({ Component, pageProps }) {
  // you can set the `isAuthLayout` boolean on each page component
  if (Component.isAuthLayout) {
    return <AuthLayout><Component {...pageProps} /></AuthLayout>
  }
  // and if `isAuthLayout` is falsey then no layout is rendered
  return <Component {...pageProps} />
}


// pages/sign-in.tsx
export default function LoginPage() {
  return // your login page code
}
LoginPage.isAuthLayout = true;


// pages/sign-up.tsx
export default function RegisterPage() {
  return // your register page code
}
RegisterPage.isAuthLayout = true;
```

**Reply by t-zander** [reply]
Sorry for late reply. It works like a charm. Thank you!
